src/main/java/uk/gov/hmcts/befta/player/DefaultBackEndFunctionalTestScenarioPlayer.java

```diff
@@ -149,6 +152,50 @@ private void prepareARequestWithAppropriateValues(BackEndFunctionalTestScenarioC
                 + JsonUtils.getPrettyJsonFromObject(scenarioContext.getTestData().getRequest()));
     }
 
+    private void runPrerequisitesSpecifiedInTheContext(final BackEndFunctionalTestScenarioContext scenarioContext)
+            throws IOException {
+
+        List<Object> prerequisites = scenarioContext.getTestData().getPrerequisites();
+
+        if (prerequisites != null && !prerequisites.isEmpty()) {
+            scenario.write("Prerequisite processing started: [" + scenarioContext.getReference() + "]");
+
+            // prerequisites as list of objects
+            for (Object prerequisite: prerequisites) {
+                // if prerequisite is just a string
+                if (prerequisite instanceof String) {
+                    String prerequisiteAsString = (String)prerequisite;
+                    runSinglePrerequisite(scenarioContext, prerequisiteAsString, prerequisiteAsString);
+
+                    // if prerequisites is a map of pairs: "reference": "guid"
+                } else if (prerequisite instanceof Map) {
+                    @SuppressWarnings("unchecked")
+                    Map<String, String> prerequisiteAsMap = (Map<String, String>) prerequisite;
+                    for (Map.Entry<String,String> prerequisiteAsEntry : prerequisiteAsMap.entrySet()) {
+                        runSinglePrerequisite(scenarioContext, prerequisiteAsEntry.getKey(), prerequisiteAsEntry.getValue());
+                    }
+
+                } else {
+                    throw new FunctionalTestException("Unrecognised prerequisite data type");
+                }
+            }
+
+            scenario.write("Prerequisite processing complete: [" + scenarioContext.getReference() + "]");
+        }
+    }
+
+    private void runSinglePrerequisite(final BackEndFunctionalTestScenarioContext scenarioContext, String testDataReference, String testDataId)
+            throws IOException {
+
+        // avoid loops by skipping references that have already been loaded
+        if (!this.scenarioContext.getChildContexts().containsKey(testDataReference)) {
+            scenario.write("Prerequisite: [" + scenarioContext.getReference() + "].[" + testDataReference + "] from [" + testDataId + "]");
+            performAndVerifyTheExpectedResponseForAnApiCall(PREREQUISITE_SPEC, testDataId, testDataReference);
+        } else {
+            scenario.write("Skipping existing prerequisite: [" + scenarioContext.getReference() + "].[" + testDataReference + "]");
```

Comment:
I need to re-think of this logic here with a fresher mind.

Reply:
Updated as agreed in workshop.   However have re-used `InvalidTestDataException` + message rather than using the proposed new `CyclicPrerequisiteDependencyException` as SonarLint was complaining of inheriiteence depth being too large.

In addition to expanding unit tests to cover changes, I have also manually tested both cyclic dependency and the new skip executed prerequisite scenarios.